packages/next/src/rsc/components/PreviewIndicator.tsx

```diff
@@ -0,0 +1,35 @@
+import { cookies, draftMode, headers } from 'next/headers';
+import React from 'react';
+import { redirect } from 'next/navigation';
+
+import { COOKIE_NAME } from '../handlers/previewRouteHandler';
+
+export type PreviewIndicatorProps = {
+	className?: string;
+};
+
+export const PreviewIndicator: React.FC<PreviewIndicatorProps> = ({ className }) => {
+	const { isEnabled } = draftMode();
+
+	if (!isEnabled) {
+		return null;
+	}
+
+	async function disableDraftMode() {
+		'use server';
+
+		const currentUrl = headers().get('x-headstartwp-current-url') ?? '/';
```

Comment:
I could have made this component a client component and get the current path with `usePathname` but this solution seems to work well where the middleware add the current URL as a custom header.

The nice thing about App Router is that so far HeadstartWP has contributed zero KB to the final app bundle.